crates/driver/src/domain/competition/solution/fee.rs

```diff
@@ -105,6 +111,25 @@ impl Fulfillment {
         }
     }
 
+    /// Returns the surplus denominated in the sell token.
+    fn surplus_in_sell_token(
```

Comment:
Sounds like this should also be a function on `Trade`. Additionally one could consider changing the signature to only take `surplus` and `prices`.
This together with the other suggestion might make the individual fee implementations more explicit.

Reply:
moved to `Fulfillment`